packages/story-editor/src/components/panels/document/publish/publish.js

```diff
@@ -137,51 +179,108 @@ function PublishPanel() {
     let message = __('No file types are currently supported.', 'web-stories');
 
     if (allowedImageFileTypes.length) {
       message = sprintf(
         /* translators: %s: list of allowed file types. */
-        __('Please choose only %s as publisher logo.', 'web-stories'),
+        __('Please choose only %s as a poster.', 'web-stories'),
         translateToExclusiveList(allowedImageFileTypes)
       );
     }
 
     return message;
   }, [allowedImageFileTypes]);
 
-  const posterErrorMessage = useMemo(() => {
+  const onPublisherLogoChange = (option) => {
+    const sizeLogo = option.media_details.sizes?.['web-stories-publisher-logo'];
+    updateStory({
+      properties: {
+        publisherLogo: {
+          id: option.id,
+          url: sizeLogo?.source_url || option.source_url,
+          width: sizeLogo?.width || option.media_details.width,
+          height: sizeLogo?.height || option.media_details.height,
+        },
+      },
+    });
+  };
+
+  const publisherLogoErrorMessage = useMemo(() => {
     let message = __('No file types are currently supported.', 'web-stories');
 
     if (allowedImageFileTypes.length) {
       message = sprintf(
         /* translators: %s: list of allowed file types. */
-        __('Please choose only %s as a poster.', 'web-stories'),
+        __('Please choose only %s as publisher logo.', 'web-stories'),
         translateToExclusiveList(allowedImageFileTypes)
       );
     }
 
     return message;
   }, [allowedImageFileTypes]);
 
+  const openMediaPicker = useMediaPicker({
+    onSelect: onPublisherLogoChange,
+    cropParams: {
+      width: 96,
+      height: 96,
+    },
+    type: allowedImageMimeTypes,
+    onSelectErrorMessage: publisherLogoErrorMessage,
+  });
+
+  const publisherLogoOptionRenderer = forwardRef(
+    ({ option: option, ...rest }, ref) => {
+      if (option.props) {
+        return option;
+      }
+      const src =
+        option.media_details.sizes?.['web-stories-publisher-logo']
+          ?.source_url || option.source_url;
+      return (
+        <Option value={option.id} ref={ref} {...rest}>
+          <LogoImg src={src} alt="" />
+        </Option>
+      );
+    }
+  );
+  const activeItemRenderer = () =>
+    publisherLogo.id ? (
+      <LogoImg src={publisherLogo.url} alt="" />
+    ) : publisherLogos.length ? (
+      <Text as="span" size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL}>
+        {__('Select logo', 'web-stories')}
+      </Text>
+    ) : (
+      <Text as="span" size={THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.SMALL}>
+        {__('No logo', 'web-stories')}
+      </Text>
+    );
+
+  const settingsLink = `${window.location.origin}/wp-admin/edit.php?post_type=web-story&page=stories-dashboard#/editor-settings`;
+
+  const publisherLogosWithUploadOption = [...publisherLogos];
+  if (hasUploadMediaAction) {
+    publisherLogosWithUploadOption.unshift(
+      <Option key="upload" onClick={openMediaPicker}>
```

Comment:
Ah, okay, now I see where the Media Picker "not closing" bug comes from. `onClick={openMediaPicker}` is used twice here -- once on the option and once on `Text`. So it gets opened twice.

Reply:
Yep, thanks for catching that.